A beginners guide to python (#1340)
section-engineering-education/engineering-education: articles/a-beginners-guide-to-python/index.md

Python is among the most popular programming languages in the world. Python is ranked high in different StackOverflow programming surveys each year. This article helps beginners familiarize themselves with the basic concepts of Python.  

### Introduction

> According to the official [documentation](https://www.python.org/), Python is a high-level and general-purpose programming language. It is a preferred programming language to many developers due to its huge range of libraries. Individuals can use these libraries to create complex applications rather than building everything from scratch. Other reasons why developers love Python is due to its simplicity, flexibility, and versatility. Today, Python is being used in upcoming fields such as machine learning and artificial intelligence.

Reviewer: ```suggestion
According to the official [documentation](https://www.python.org/), Python is a high-level and general-purpose programming language. It is a preferred programming language for many developers due to its wide range of libraries. Individuals can use these libraries to create complex applications rather than building everything from scratch. Other reasons why developers love Python is due to its simplicity, flexibility, and versatility. Today, Python is being used in upcoming fields such as machine learning and artificial intelligence.
```

### What You will learn

In this tutorial, you will learn how to define and use the following elements
•   Variables
•   Methods
•   loops
•   lists and arrays
•   Classes

### Prerequisites

> For simplicity, we will be using IDLE in this tutorial. IDLE is the default python code editor and compiler. You can download it from [here](https://www.python.org/downloads/).

Reviewer: ```suggestion
For simplicity, we will be using IDLE (Integrated Development and Learning Environment) in this tutorial. IDLE is the default python code editor and compiler. You can download it from [here](https://www.python.org/downloads/).
```

Also, note that this tutorial is based on python 3.9 rather than the 3.6 version.

### Understanding variables
Variables are simply declarations that are used to store certain values. For instance, the variable `name` can hold the value of `John Smith.` Several rules need to be considered when declaring variable names. For starters, a variable name cannot begin with a number. 

2name  = incorrect #incorrect

name = correct  #correct

Variable names are case sensitive. This means that the variable `school` is not the same as `School`.

Variables can hold different data types. This includes strings, integers, Booleans, long, lists, and arrays.

In Python, we do not need to declare the data type while writing a variable. This is because the code is compiled and interpreted later. The compiler will throw an error in case there is a mismatch in the data types.

Let’s talk about the different data types.

1. Strings

Strings are usually presented in a text format. We declare a string variable, as shown below.

> ```

Reviewer: Please add syntax highlighting to all the code snippets in this article. For example:

```md
```python
print("Hello")
```

name = “john”
school = “Alliance Francaise”

> ```

Reviewer: Here, you can also define how multiline strings can be declared. For reference:

https://www.w3schools.com/python/python_strings.asp

When we run `print(name)`, the output will be `john`.

2. Integers

These variables hold numeric values, as shown below.
```

> math=90

Reviewer: In this code block, you have inconsistently used spacing between "equal to" and the variable value. Please fix it.

chemistry = 100
biology =70

> ```

Reviewer: You can also give an example of declaring an integer, a string and adding both of them.

Author: @cyberShaw could explain this comment. I think declaring a string and integer is covered. Thanks

Reviewer: Yeah, but you could define what would happen when a string and an integer are added.

For example,

```py
var1 = "30"
var2 = 20

print(var1+var2)
```

Author: done

We can find the total of the above variables using the following statement.
```
print(math+chemistry+biology)
```
The total is `260`.

3. Booleans

> There are only two Boolean values: `True` and `False`. In other words, something can either be true or false. We declare these values, as shown below. Please note that Python is case sensitive.

Reviewer: You can discuss how what would be returned by the following statements:

```py
bool("abc")
bool(0)
```

```python
isOn=True
isChecked=False
```

4. Float

This data type consists of numbers that have a decimal place. A perfect example of a float variable is highlighted below.

```
Bmi = 45.7
```

### Understanding lists
Lists allow us to store numerous elements in a particular variable. For instance, we can have a list that stores all the student names in a class. We use `[]` to define a list.

> ```

Reviewer: You can also discuss how lists can be defined using the list() function.

students = [] #list example
```

Elements in a list are usually separated by a comma, as shown below.
```
students = [“john”, “Mary Thomas”, “John Smith”]
```

> Each element in the above `students` list has an index. By default, the first index is 0. So the item at index [0] is `john`, while the value at index `1` is `Mary Thomas`. 

Reviewer: Please illustrate an example of negative list indices.

A list of integers will look as follows.

```
student_marks =[90, 78, 90, 78]
```
We can access different list functionalities using built-in functions. For instance, to add a value to the `student_marks` list, we use the `append` function.

```python

> student_marks.append(“Guardian Angel”)

Reviewer: Include a print() statement and show what the output of this list would be

```

The above function adds `Guardian Angel` at the end of the `student_marks` list.
We use `len(student_marks)` to determine the length of the list.

> We use the remove() function to delete something from the list.

Reviewer: ```suggestion
We use the `remove()` function to delete something from the list.
```

> We use the remove() function to delete something from the list.

Reviewer: Illustrate the remove() function with an example.

### Understanding functions or methods
Methods are quite critical in programming. They help store reusable code. This means that a person can call already declared methods rather than writing statements from scratch repeatedly. This saves significant time, which can be invested in other productive activities.

In Python, we use the `def` keyword to declare a function. 
An example of a python method is shown below.
```
def readData():
    print('success')
```
The above function prints `success` when it’s invoked.
We can also pass data to a method, perform some calculations, and return the results. This is demonstrated in the code snippet below.

```
def calculateTotal(chem, bio):
    return chem+bio

> print(calculateTotal (90,80)

Reviewer: ```suggestion
print(calculateTotal(90,80))
```

```
The `calculateTotal` method takes in two parameters (chem, bio). The function then returns the sum of the two values. It is important to take note of the data types when passing parameters. For instance, the `calculateTotal` method will not work when we pass in a string as a parameter. This is because the program cannot sum up an integer and a string. 
As shown above, we can call the `calculateTotal` method directly from our print statement.
```

> print(calculateTotal (90,80)

Reviewer: ```suggestion
print(calculateTotal(90,80))
```

```
The `return` keyword ensures that the method returns a result after execution.

Note that a function can also call another method. This is illustrated below.

```python
def readData(chem, bio):
    return chem+bio

def getTotal():
    print(readData(90,80)) #calls the readData method

getTotal()
```

### Understanding loops
Loops are critical because they allow us to iterate through lists, check for different conditions, and continuously execute various statements. The main loops are `for` and `while`.

1. For loops

> As noted, we can use a for loop to iterate through a list, as shown below.

Reviewer: When you're going to define an example, Please end the line with a colon.

```suggestion
As noted, we can use a for loop to iterate through a list, as shown below:
```

```

> Student_list=[“John Doore”,”Matu Smith”]

Reviewer: ```suggestion
student_list=[“John Doore”,”Matu Smith”]
```

This should be student_list with a lowercase `s`, since variables in Python are case sensitive.

for x in student_list:
        print(x)
```
The above `for` loop will print every item in the student_list.

2. While loops
A while can help us check for a particular condition. For instance, while something is true specific statements can be executed. Here is an example of a while loop in action.

```
isChecked=false
while isChecked == true:
  print('Hallo there')
```
>Note that the above while loop will be executed indefinitely until isChecked is set to false. You can press ctrl+c to stop the loop.

### Classes

Classes are a vital component of object-oriented programming. When creating a class, you must use the `class` keyword. Other elements are then nested in the class. Here is an example of a python class.

```python
class Farmer: # a class with the name farmer
    name="John" # A variable
    produce ="1000kgs" # A variable

farmer=Farmer() #instatiating the class as an object. 
print(farmer.name) # accessing the properties of the Farmer class.
```
Classes can help as group things with similar characteristics. 

We can also assign values to class variables using the `init` function.

```python
class Farmer:

>   def __init__(self, farmername, produce):

Reviewer: Since you've discussed classes, please add a small explanation about how the `self` keyword works.

    self.farmername = farmername
    self.produce = produce

farmer = Farmer("Carry Sminson", "10,000kgs")

print(farmer.farmername, farmer.produce)
```

You can also declare a method in a class and use it later, as shown below.

```python
class Farmer:
  def __init__(self, farmername, produce):
    self.farmername = farmername
    self.produce = produce

  def printDetails(self): # Method
      print(self.farmername, self.produce)

farmer = Farmer("Carry Sminson", "10,000kgs")

farmer.printDetails()
```

### Conclusion

> We have learned how to define and use variables, methods, classes, loops, and lists from this tutorial. You can learn more about other concepts from [here](https://docs.python.org/3/). The secret to becoming a python exert is to practice regularly. 

Reviewer: ```suggestion
We have learned how to define and use variables, methods, classes, loops, and lists from this tutorial. You can learn more about other concepts from [here](https://docs.python.org/3/). The secret to becoming a python expert is to practice regularly. 
```
